pages/api/class/index.ts

```diff
@@ -5,21 +5,29 @@ import { ClassInput, ClassTranslationInput } from "@models/Class";
 import { validateClassData } from "@utils/validation/class";
 import { getClassInfoWithProgramId } from "@database/program-card-info";
 import { stripe } from "services/stripe";
+import { getSession } from "next-auth/client";
+import { isAdmin } from "@utils/session/authorization";
 
 /**
  * handle controls the request made to the class resource
  * @param req API request object
  * @param res API response object
  */
 export default async function handle(req: NextApiRequest, res: NextApiResponse): Promise<void> {
+    const session = await getSession({ req });
     switch (req.method) {
         case "GET": {
             const { id: programId, archived } = req.query;
 
+            const archivedParsed = Boolean(JSON.parse((archived as string) || "false"));
```

Comment:
@MatoPlus Can you explain this part to me? How come we're doing this before we check `!isAdmin(session)`?

Reply:
My understanding is that the route is open to the public, but archived items requires admin access.

Reply:
Might be good to leave a comment here or perhaps make sure this is documented somewhere along with which other routes should be private

Reply:
i'll make the comment a bit more inutitive, we're just following the least privilege access principle for our api routes.